### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by apt2tell**
## Adding reducers and actions

> in this feature, the following tasks were completed:

- Added redux folder
- Added books folder and books.js file
- Created categories folder and categories.js file
- Created configureStore.js
- Added reducer  functions and action creators for books and categories respectively

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const initialState = [];
```
Comment (danmainah) [inline src/redux/categories/categories.js:3]: - [ ] @Okechukwu-muokwugwo great work in your redux, you forgot to include the action type for checking your status, kindly ensure it follows the duck pattern, as this is a project requirement.
- [ ] image

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const checkStatus = () => ({
+  type: 'CHECK_STATUS',
+});
+
+const categoriesReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'CHECK_STATUS':
+      return [...state, 'IN_PROGRESS'];
```
Comment (danmainah) [inline src/redux/categories/categories.js]: - [ ] Kindly check on your returned string, it's supposed to return the "Under construction" string, this is a project requirement.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Okechukwu-muokwugwo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] [optional] It would be nice if you made your project description in the readme file to describe the current status of your application including the resent added fucntionality.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Okechukwu-muokwugwo 😃,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the requested changes from the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**